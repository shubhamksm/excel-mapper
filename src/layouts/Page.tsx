import { Button, ButtonProps } from "@/components/ui/button";

type PageControlButtonProps = Omit<ButtonProps, "style">;

export type PageProps = {
  title: string;
  mainContent: JSX.Element;
  previousLabel?: string;
  nextLabel?: string;
  nextButtonProps?: PageControlButtonProps;
  previousButtonProps?: PageControlButtonProps;
};

const Page = ({
  title,
  mainContent,
  nextLabel,
  nextButtonProps,
  previousLabel,
  previousButtonProps,
}: PageProps) => {
  return (
    <div className="w-full h-full flex flex-col items-center gap-y-4">
      <h1 className="text-2xl">{title}</h1>
      <div className="flex-grow h-full w-full overflow-auto">{mainContent}</div>
      <div className="w-full">
        {previousLabel && (
          <Button style={{ float: "left" }} {...previousButtonProps}>
            {previousLabel}
          </Button>
        )}
        {nextLabel && (
          <Button style={{ float: "right" }} {...nextButtonProps}>
            {nextLabel}
          </Button>
        )}
      </div>
    </div>
  );
};

export default Page;
