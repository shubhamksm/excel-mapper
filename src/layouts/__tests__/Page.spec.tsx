import { render, screen } from "@testing-library/react";
import Page from "../Page";

describe("Page Layout", () => {
  it("Should match the snapshot", () => {
    const { container } = render(
      <Page
        title="Test Title"
        nextLabel="Next Label"
        previousLabel="Previous Label"
      >
        <span>Test Content</span>
      </Page>
    );

    expect(container).toMatchSnapshot();
  });

  it("Should render all content", () => {
    render(
      <Page
        title="Test Title"
        nextLabel="Next Label"
        previousLabel="Previous Label"
      >
        <span>Test Content</span>
      </Page>
    );
    expect(screen.getByText("Test Title")).toBeInTheDocument();
    expect(screen.getByText("Next Label")).toBeInTheDocument();
    expect(screen.getByText("Previous Label")).toBeInTheDocument();
  });
});
