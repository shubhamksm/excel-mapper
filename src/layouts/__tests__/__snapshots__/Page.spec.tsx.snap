// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Layout Should match the snapshot 1`] = `
<div>
  <div
    class="w-full h-full flex flex-col items-center gap-y-4"
  >
    <h1
      class="text-2xl"
    >
      Test Title
    </h1>
    <div
      class="flex-grow h-full w-full overflow-auto"
    >
      <span>
        Test Content
      </span>
    </div>
    <div
      class="w-full"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
        style="float: left;"
      >
        Previous Label
      </button>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
        style="float: right;"
      >
        Next Label
      </button>
    </div>
  </div>
</div>
`;
